Calibration/HcalCalibAlgos/test/HcalLaserTest.cc

```diff
@@ -0,0 +1,226 @@
+  minADCHBHE_(config.getParameter<int>("minADCHBHE")),
+  minADCHF_(config.getParameter<int>("minADCHF")),
+  testMode_(config.getUntrackedParameter<bool>("testMode", false)),
+  eventsByType_(),
+  passedEventsByType_() { 
+
+  usesResource(TFileService::kSharedResource);
+}
+
+HcalLaserTest::~HcalLaserTest() { }
+ 
+void HcalLaserTest::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("InputHBHE",edm::InputTag("source"));
+  desc.add<edm::InputTag>("InputHF",edm::InputTag("source"));
+  desc.add<edm::InputTag>("UMNioDigis",edm::InputTag("UMNioDigis"));
+  desc.add<int>("minADCHBHE",10);
+  desc.add<int>("minADCHF",10);
+  desc.add<double>("minFracDiffHBHELaser",0.3);
+  desc.add<double>("minFracHFLaser",0.3);
+  desc.addUntracked<bool>("testMode",false);
+  descriptions.add("hcalLaserTest",desc);
+}
+
+void HcalLaserTest::analyze(edm::Event const& iEvent, 
+			    edm::EventSetup const& iSetup) {
+  
+  edm::Handle<HBHEDigiCollection>     hbhe_digi; 
+  iEvent.getByToken(inputTokenHBHE_,  hbhe_digi);
+
+  edm::Handle<QIE10DigiCollection>    hf_digi; 
+  iEvent.getByToken(inputTokenHF_,    hf_digi);
+
+  // Count digis in good, bad RBXes.  ('bad' RBXes see no laser signal)
+  double badrbxfracHBHE(0), goodrbxfracHBHE(0), goodrbxfracHF(0);
+  int NbadHBHE = 72*3;             // 3 bad RBXes, 72 channels each
+  int NgoodHBHE= 2592*2-NbadHBHE;  // remaining HBHE channels are 'good'
+  int NgoodHF  = 864*4;
+  int eventType = 0;
+  int laserType = 0;
+
+  // Verify proper functioning on calibration stream events, 
+  // where the laser firing location "laserType"
+  // is tagged by the UMNio board for eventType==14
+
+  edm::Handle<HcalUMNioDigi> cumn;
+  iEvent.getByToken(ioTokenUMN_, cumn);
+
+  eventType = static_cast<int>(cumn->eventType());
+  laserType = static_cast<int>(cumn->valueUserWord(0));
+
+  if (eventType == 14) eventsByType_.at(laserType)++;
+
+  if (testMode_)
+    edm::LogVerbatim("HcalLaserTest") 
+      << "hbhe digi collection size: " << hbhe_digi->size() << "\n"
+      << "hf digi collection size: " << hf_digi->size() << "\n"
+      << "Event type: " << eventType 
+      << " Laser type: " << laserType << std::endl;
+
+  for (HBHEDigiCollection::const_iterator hbhe = hbhe_digi->begin();  
+       hbhe != hbhe_digi->end(); ++ hbhe){
+    const HBHEDataFrame digi = (const HBHEDataFrame)(*hbhe);
+    HcalDetId myid=(HcalDetId)digi.id();
+    bool isbad(false); 
+
+    bool passCut(false);
+    int  maxdigiHB(0);
+    for (int i=0; i<digi.size(); i++) 
+      if(digi.sample(i).adc() > maxdigiHB) maxdigiHB = digi.sample(i).adc();
+    if (maxdigiHB > minADCHBHE_) passCut = true;
+
+    if ( myid.subdet()==HcalBarrel && myid.ieta()<0 ) {
+      if      (myid.iphi()>=15 && myid.iphi()<=18) isbad=true;
+      else if (myid.iphi()>=27 && myid.iphi()<=34) isbad=true;
+    }
+
+    if (isbad) h_hb2_->Fill(maxdigiHB);
```

Comment:
I am adding a helper class. I shall later on update the HLT class once this PR is accepted and integrated.